common/serde-reflection/src/format.rs

```diff
@@ -103,7 +103,7 @@ pub enum VariantFormat {
 
 pub trait FormatHolder {
     /// Visit all the formats in `self` in a depth-first way.
-    fn visit(&self, f: &mut dyn FnMut(&Format) -> Result<()>) -> Result<()>;
+    fn visit<'a>(&'a self, f: &mut dyn FnMut(&'a Format) -> Result<()>) -> Result<()>;
```

Comment:
What's the reason why these lifetime annotations are needed?

Reply:
Fair question. I updated the last commit with a better unit test to show the difference.